bftengine/tests/simpleTest/config.cpp

```diff
@@ -67,241 +62,28 @@ const uint16_t basePort = 3710;
-
-  std::set <pair<uint16_t, string>> publicKeysOfReplicas;
-
-  // read from publicKeysFile
-
-  for (int i = 0; i < numOfReplicas; i++) {
-    fscanf(publicKeysFile, "%" MAX_ITEM_LENGTH_STR "s\n", tempString);
-
-    string name(tempString);
-
-    string expectedName = string("replica") + std::to_string(i);
-
-    if (expectedName != name) return false;
-
-    fscanf(publicKeysFile, "%" MAX_ITEM_LENGTH_STR "s\n", tempString);
-
-    string publicKeyString(tempString);
-
-    publicKeysOfReplicas.insert({i, publicKeyString});
+// Create a replica config for the replica with index `replicaId`.
+// inputReplicaKeyfile is used to read the keys for this replica, and default
+// values are loaded for non-cryptographic configuration parameters.
+void getReplicaConfig(uint16_t replicaId, ReplicaConfig* outConfig) {
+  
+  std::string keyfileName = namesOfKeyfiles[replicaId];
+  std::ifstream keyfile(keyfileName);
+  if (!keyfile.is_open()) {
+    throw std::runtime_error("Unable to read replica keyfile.");
   }
+  
+  bool succ = inputReplicaKeyfile(keyfile, keyfileName, *outConfig);
+  if (!succ)
+    throw std::runtime_error("Unable to parse replica keyfile.");
 
-  // Read threshold signatures
-  IThresholdFactory* slowCommitFactory = nullptr;
-  IThresholdSigner* thresholdSignerForSlowPathCommit = nullptr;
-  IThresholdVerifier* thresholdVerifierForSlowPathCommit = nullptr;
-
-  IThresholdFactory* optimisticCommitFactory = nullptr;
-  IThresholdSigner* thresholdSignerForOptimisticCommit = nullptr;
-  IThresholdVerifier* thresholdVerifierForOptimisticCommit = nullptr;
-
-  slowCommitFactory = createThresholdFactory(publicKeysFile);
-  thresholdVerifierForSlowPathCommit =
-      createThresholdVerifier(publicKeysFile,
-                              numOfReplicas - fVal,
-                              numOfReplicas,
-                              slowCommitFactory);
-
-  optimisticCommitFactory = createThresholdFactory(publicKeysFile);
-  thresholdVerifierForOptimisticCommit =
-      createThresholdVerifier(publicKeysFile,
-                              numOfReplicas,
-                              numOfReplicas,
-                              optimisticCommitFactory);
-
-  // read from privateKeysFile
-
-  int tempInt = 0;
-  fscanf(privateKeysFile, "%d\n", &tempInt);
-  if (tempInt != replicaId) return false;
-
-  fscanf(privateKeysFile, "%" MAX_ITEM_LENGTH_STR "s\n", tempString);
-
-  string sigPrivateKey(tempString);
-
-  thresholdSignerForSlowPathCommit =
-      createThresholdSigner(privateKeysFile, replicaId+1, slowCommitFactory);
-
-  thresholdSignerForOptimisticCommit =
-      createThresholdSigner(privateKeysFile,
-                            replicaId+1,
-                            optimisticCommitFactory);
-
-  delete slowCommitFactory;
-  delete optimisticCommitFactory;
-
-  // set configuration
+  // set non-cryptographic configuration
 
-  outConfig->replicaId = replicaId;
-  outConfig->fVal = fVal;
```

Comment:
I noticed that, but I was kind of on the fence about whether to remove `fVal` from `config.cpp` or not. I noticed it now caused compile-time warnings because I had removed the statements where it was being used, but it also struck me as a possibly important definition because it is a fundamental parameter to the Concord deployment, and I am not confident `simpleTest` might assume it has 4 replicas and F = 1 elsewhere. 

Thank you for your thoughts on this, I will remove the `fVal` constant from `config.cpp`.